### PR TITLE
Remove standby.signal if the data is backuped from standby

### PR DESCRIPTION
**Pull request by mikecaat**
Now, pg_rman remains standby.signal if exists, which leads that the postgresql server starts as standby if the recovery target is latest. But, this is not expected behavior because it doesn't treat the backup as restoring on standby automatically, which is documented now.

So, this patch removes the standby.signal if exits.
This patch must be applied to REL_12_STABLE and REL_13_STABLE branches.

### CODE REVIEW

**Comment by MoonInsung** [comment]
Dear mikecaat.

Thank you for responding to the issue.
I agree with your opinion, but there are a few comments.
Comments will be written directly to the PR code on GitHub.

Best regards.
Moon.

File: restore.c [hunk 1]
```diff
@@ -799,6 +806,33 @@ create_recovery_signal(void)
...
+		snprintf(path, lengthof(path), "%s/standby.signal", pgdata);
+
+		if (stat(path, &stat_buf) == 0)
+		{
+			if (remove(path))
+			{
+				elog(ERROR, _("could not remove \"%s\": %s"), path,
+					strerror(errno));
+			}
+			elog(INFO, _("removed standby.signal"));
```
Comment (MoonInsung) [inline restore.c:831]: How about adding a little more warning message(info level)?
For example, "SR configuration has been released and operates in standalone mode. If SR configuration is required, a new setting is required." I think it would be better to add a message like this.
It would be a think to inform users who did not expect this behavior "Why that released the SR configuration."

File: sql/backup_from_standby.sh [hunk 2]
```diff
@@ -203,9 +222,31 @@ ${BACKUP_RESTORE_COMMAND}
 ${BACKUP_TARGET_XID_COMMAND}
 EOF
 pg_ctl start -w -t 600 -D ${PGDATA_PATH} > /dev/null 2>&1
-psql -p ${TEST_PGPORT} --no-psqlrc -d pgbench -c "SELECT * FROM pgbench_branches;" > ${TEST_BASE}/TEST-0002-after.out
+get_database_data > ${TEST_BASE}/TEST-0002-after.out
 diff ${TEST_BASE}/TEST-0002-before.out ${TEST_BASE}/TEST-0002-after.out
+echo ''
 
+# check to start as primary (stand-alone) if the recovery target is latest
```
Comment (MoonInsung) [inline sql/backup_from_standby.sh:229]: Wouldn't it be better to perform a restore on the standby side for this regresstion test?
I think the goal of this regression test is to restore the database cluster where standby.signal exists, the restore from the primary side does not seem to be related to your opinion.

**Review by MoonInsung — COMMENTED** [review]
Dear mikecaat.

Thank you for responding to the issue.
I agree with your opinion, but there are a few comments.
Comments will be written directly to the PR code on GitHub.

Best regards.
Moon.

File: restore.c (hunk 1 above)
Comment (mikecaat) [reply restore.c:831]: Yes, I agree. I'll change it

File: sql/backup_from_standby.sh (hunk 2 above)
Comment (mikecaat) [reply sql/backup_from_standby.sh:229]: Thanks for your comments! 

Sorry, I couldn't understand yet. Although I agree with your comments, I thought the test is to perform a restore using standby-side data?

1. start primary and standby
2. backup from standby side
3. stop primary
4. restore the data directory using the backup data got in the 2nd phase
5. start using the data directory

Do you mean that data directory must be standby server's one in the 4th and the 5th phase? For example, is it better to change ${PGDATA_PATH} to ${SBYDATA_PATH}?
> pg_rman restore -B ${BACKUP_PATH} -D ${PGDATA_PATH} --quiet;echo $?
> sed -i -e "s/^port = .*$/port = ${TEST_PGPORT}/" ${PGDATA_PATH}/postgresql.conf
> pg_ctl start -w -t 600 -D ${PGDATA_PATH} > /dev/null 2>&1

If I'm missing something, please let me know.

**Review by mikecaat — COMMENTED** [review]
Thanks for your comments!

File: sql/backup_from_standby.sh (hunk 2 above)
Comment (MoonInsung) [reply sql/backup_from_standby.sh:229]: Oh... I'm sorry.
I imagined that using the backup file acquired from the standby side would be restored on the primary and standby sides, thought that a regression test was necessary.
In other words, my imagined "pg_ctl START -d SBYDATA_PATH" after copying the restored database cluster to "SBYDATA_PATH."

However, the pg_rman manual wrote.
"Basically, the backup from standby-site is used for restoring on MASTER. pg_rman doesn't treat the backup as restoring on standby automatically." 
In other words, it means "basically does not support restore from standby", so I don't think you need to add this test.

In other words, please ignore the comments on the regression test that I have requested.
Please apologize for making such a strange comment.

**Review by MoonInsung — COMMENTED** [review]
Thank you!

**Comment by MoonInsung** [comment]
Back-patch to REL_13_STABLE and REL_12_STABLE branches.
